File: app/src/main/java/com/xmx/tango/common/log/OperationLog.java
package com.xmx.tango.common.log;

import android.content.ContentValues;
import android.database.Cursor;

import com.xmx.tango.common.data.sql.ISqlEntity;

import java.util.Date;

/**
 * Created by The_onE on 2016/9/4.
 */
public class OperationLog implements ISqlEntity {
    public long mId = -1;
    public String mOperation;
    public Date mTime;

    @Override
    public String tableFields() {
        return "ID integer not null primary key autoincrement, " +
                "Operation text, " +
                "Time integer not null default(0)";
    }

    @Override
    public ContentValues getContent() {
        ContentValues content = new ContentValues();
        if (mId > 0) {
            content.put("ID", mId);
        }
        content.put("Operation", mOperation);
        content.put("Time", mTime.getTime());
        return content;
    }

    @Override
    public OperationLog convertToEntity(Cursor c) {
        OperationLog entity = new OperationLog();
        entity.mId = c.getLong(0);
        entity.mOperation = c.getString(1);
        entity.mTime = new Date(c.getLong(2));

        return entity;
    }
}

File: app/src/main/java/com/xmx/tango/common/log/OperationLog.kt
package com.xmx.tango.common.log

import android.content.ContentValues
import android.database.Cursor

import com.xmx.tango.common.data.sql.ISqlEntity

import java.util.Date

/**
 * Created by The_onE on 2016/9/4.
 * 操作日志实体
 */
class OperationLog : ISqlEntity {
    private var mId: Long = -1
    var mOperation: String? = null
    var mTime = Date()

    override fun tableFields(): String {
        return "ID integer not null primary key autoincrement, " +
                "Operation text, " +
                "Time integer not null default(0)"
    }

    override fun getContent(): ContentValues {
        val content = ContentValues()
        if (mId > 0) {
            content.put("ID", mId)
        }
        content.put("Operation", mOperation)
        content.put("Time", mTime.time)
        return content
    }

    override fun convertToEntity(c: Cursor): OperationLog {
        val entity = OperationLog()
        entity.mId = c.getLong(0)
        entity.mOperation = c.getString(1)
        entity.mTime = Date(c.getLong(2))

        return entity
    }
}
